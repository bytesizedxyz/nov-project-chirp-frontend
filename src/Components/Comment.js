import React from 'react';
import './Post.css';

export const Comment = () => {
  return (
    <div className="comment-body">
      <span>
        <span className="comment-header">
          <img
            className="comment-profile-image"
            src="https://www.neweurope.eu/wp-content/uploads/2018/02/h_53880267.jpg"
            alt="A user visual identifier."
          />
          <h4>Users Name:</h4>
          <p>Hi, I'm a comment!</p>
        </span>
        <span className="reply-bar">
          <p>Like</p>
          <p>Hate</p>
          <p>Reply</p>
        </span>
      </span>

      <span className="reply-input">
        <input type="text" />
        <p>Send</p>
      </span>

      {/* <span className="reply-header">
        <img
          className="reply-profile-image"
          src="https://www.neweurope.eu/wp-content/uploads/2018/02/h_53880267.jpg"
          alt="A user visual identifier."
        />
        <h4>Users Name:</h4>
        <p>Hi, I'm a comment!</p>
      </span> */}
    </div>
  );
};
