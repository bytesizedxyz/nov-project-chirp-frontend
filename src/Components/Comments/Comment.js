import React, { Component } from "react";
import "../Post/Post.css";

class Comment extends Component {
  state = {
    comments: [{ userName: "A User", comment: "I WIN" }],
    commentValue: ""
  };

  handleChange = e => {
    console.log("handling comment change");
    this.setState({ commentValue: e.target.value });
  };

  submitComment = () => {
    console.log("submitting comment");
    const user = "fake user";
    const { commentValue } = this.state;
    const newComment = { userName: user, comment: commentValue };
    this.setState({ comments: this.state.comments.concat(newComment) });
  };

  render() {
    const { comments, commentValue } = this.state;
    return (
      <div className="comment-body">
        <span data-testid="comments">
          {comments.map((comment, index) => {
            return (
              <span key={index} className="comment-header">
                <img
                  className="comment-profile-image"
                  src="https://www.neweurope.eu/wp-content/uploads/2018/02/h_53880267.jpg"
                  alt="A user visual identifier."
                />
                <h4>{comment.userName}</h4>|<p>{comment.comment}</p>
              </span>
            );
          })}
          <span className="reply-bar">
            <p>Like</p>
            <p>Hate</p>
            <p>Reply</p>
          </span>
        </span>

        <span className="reply-input">
          <input
            placeholder="New Comment"
            value={commentValue}
            onChange={this.handleChange}
            type="text"
          />
          <p onClick={this.submitComment}>Send</p>
        </span>
      </div>
    );
  }
}

export default Comment;
