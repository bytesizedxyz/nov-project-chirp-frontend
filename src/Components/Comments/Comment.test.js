import React from "react";
import { render, fireEvent, wait } from "react-testing-library";
import Comment from "./Comment";

describe("Comment functionality", () => {
  it("Can render comments", () => {
    //making comment render
    const { getByText } = render(<Comment />);

    //what to search for in rendered comment
    const firstText = getByText("I WIN");

    //expecting this to be in the renderer
    expect(firstText).toHaveTextContent("I WIN");
  });
  it("can post new comments", () => {
    //making comment render
    const { getByText, getByTestId, getByPlaceholderText, asFragment } = render(<Comment />);
    //creating a snapshot to compare
    const firstSnapshot = asFragment();
    //what to search for in rendered comment
    const input = getByPlaceholderText("New Comment");
    const send = getByText("Send");
    const comments = getByTestId("comments");

    //expected to be true
    expect(send).toHaveTextContent("Send");
    expect(input).toBeEmpty();
    expect(comments.children.length).toEqual(2);

    fireEvent.change(input, { target: { value: "a new comment" } });
    expect(input.value).toBe("a new comment");

    //expecting a different snapshot
    fireEvent.click(send);
    //comparing the two snapshots and checking the length of comments
    expect(comments.children.length).toEqual(3);
    expect(firstSnapshot).toMatchDiffSnapshot(asFragment());
  });
});
