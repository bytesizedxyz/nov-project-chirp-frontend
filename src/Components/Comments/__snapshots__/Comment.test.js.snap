// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment functionality can post new comments 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -20,10 +20,26 @@
          <p>
            I WIN
          </p>
        </span>
        <span
+         class=\\"comment-header\\"
+       >
+         <img
+           alt=\\"A user visual identifier.\\"
+           class=\\"comment-profile-image\\"
+           src=\\"https://www.neweurope.eu/wp-content/uploads/2018/02/h_53880267.jpg\\"
+         />
+         <h4>
+           fake user
+         </h4>
+         |
+         <p>
+           a new comment
+         </p>
+       </span>
+       <span
          class=\\"reply-bar\\"
        >
          <p>
            Like
          </p>
@@ -39,11 +55,11 @@
        class=\\"reply-input\\"
      >
        <input
          placeholder=\\"New Comment\\"
          type=\\"text\\"
-         value=\\"\\"
+         value=\\"a new comment\\"
        />
        <p>
          Send
        </p>
      </span>"
`;
